skiko/src/commonMain/kotlin/org/jetbrains/skia/CodePoint.kt

```diff
@@ -0,0 +1,51 @@
+package org.jetbrains.skia
+
+// Copied from CharHelpers.skiko.kt
+// TODO Remove once it's available in common stdlib https://youtrack.jetbrains.com/issue/KT-23251
+internal typealias CodePoint = Int
+
+// Copied from CharHelpers.skiko.kt
+/**
+ * The minimum value of a supplementary code point, `\u0x10000`.
+ */
+private const val MIN_SUPPLEMENTARY_CODE_POINT: Int = 0x10000
+
+// Copied from CharHelpers.skiko.kt
+/**
+ * Converts a surrogate pair to a unicode code point.
+ */
+internal fun toCodePoint(high: Char, low: Char): CodePoint =
+    (((high - Char.MIN_HIGH_SURROGATE) shl 10) or (low - Char.MIN_LOW_SURROGATE)) + MIN_SUPPLEMENTARY_CODE_POINT
+
+// Copied from CharHelpers.skiko.kt
+internal fun Int.charCount(): Int = if (this >= MIN_SUPPLEMENTARY_CODE_POINT) 2 else 1
+
+internal val String.codePointsAsIntArray: IntArray
+    get() = codePoints.toList().toIntArray()
```

Comment:
Doesn't it create an extra copy?

Reply:
It does (actually more than once, inside `toList()`), but it's not easy to avoid this copy, as the number of codepoints is not known upfront. And if we just return `List<Int>` here, the callers will still need to convert it to an int array.